analyses/molecular-subtyping-embryonal/01-samples-to-subset.Rmd

```diff
@@ -112,32 +149,39 @@ fusion_biospecimens <- c(
 
 ### Disease labels
 
-First subset to embryonal **tumors**, excluding any derived cell lines.
+First subset to solid **tumors**, excluding any derived cell lines.
 
 ```{r}
 embryonal_df <- histologies_df %>%
-  filter(str_detect(broad_histology, "Embryonal tumor"),
-         sample_type == "Tumor",
+  filter(sample_type == "Tumor",
          composition == "Solid Tissue")
 ```
 
-Drop medulloblastoma and ATRT samples using the `pathology_diagnosis` labels.
+Now filter to include and exclude using the `pathology_diagnosis` and `pathology_free_text_diagnosis` labels.
 
 ```{r}
 embryonal_df <- embryonal_df %>%
-  filter(!(pathology_diagnosis %in% c("Medulloblastoma",
-                                   "Atypical Teratoid Rhabdoid Tumor (ATRT)")))
+  # Inclusion on the basis of strings in pathology_diagnosis and 
+  # pathology_free_text_diagnosis
+  filter(str_detect(str_to_lower(pathology_diagnosis), 
+                    paste0(path_dx_list$include_path_dx, collapse = "|")) |
+           pathology_diagnosis == "Other" &&
+           str_detect(str_to_lower(pathology_free_text_diagnosis), 
+                      paste0(path_dx_list$include_free_text, collapse = "|")) |
+           # "neuroblastoma" will also need to be detected in the `pathology_diagnosis` column, but we need to ensure that the `primary_site` != "Other Locations NOS" and the `pathology_free_text_diagnosis` column does note contain "peripheral"
+         str_detect(str_to_lower(pathology_diagnosis), "neuroblastoma") & !(str_detect(str_to_lower(primary_site), "other locations nos")) & !(str_detect(str_to_lower(pathology_free_text_diagnosis), "peripheral"))
+  )
```

Comment:
This code block is a lot! And that makes sense because there are a lot of considerations in what has been described for this issue and what I described in another comment in this pull request. You could instead create three separate data frames - one of each condition for inclusion - and bind them together and filter to unique entries. That should accomplish the same thing, but be easier to follow. You can see an example of this approach in the HGG filtering step in what's open on #786 (I'm permalinking to the branch): https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/37a07d2657d92bd1a06168c0a514642d7f9566b6/analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R#L121